### PR TITLE
Fix emitted code for variable statement with no bound variables

### LINKED ISSUE #3893: Empty destructuring pattern emits invalid JS
The following code:
```typescript
var a: any;

({} = a);
([] = a);

({}) => { };
([]) => { };
```

Outputs empty lists:
```
var a;
(, a);
(, a);
(function (_a) {
    var ;
});
(function (_a) {
    var ;
});
```

### PR DESCRIPTION
**Pull request by DanielRosenwasser**
Fixes #3893.

### CODE REVIEW

File: tests/cases/conformance/es6/destructuring/emptyVariableDeclarationBindingPatterns02_ES5.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// @target: es5
+// @declaration: true
```
Comment (mhegazy) [inline tests/cases/conformance/es6/destructuring/emptyVariableDeclarationBindingPatterns02_ES5.ts]: can you also add a test for assignments,

```ts
var a: any;

({} = a);
([] = a);

({}) => { };
([]) => { };
```

Comment (DanielRosenwasser) [inline tests/cases/conformance/es6/destructuring/emptyVariableDeclarationBindingPatterns02_ES5.ts]: I've already written tests for that elsewhere. See

* `tests/cases/conformance/es6/destructuring/emptyAssignmentPatterns01_ES5.ts`
* `tests/cases/conformance/es6/destructuring/emptyObjectBindingPatternParameter01.ts`
* `tests/cases/conformance/es6/destructuring/emptyArrayBindingPatternParameter01.ts`

**Comment by JsonFreeman** [comment]
Does this work for parameters?
```ts
function f({} = {}) { }
function g([] = []) { }
```

**Comment by DanielRosenwasser** [comment]
We have (`tests/cases/conformance/es6/destructuring/emptyObjectBindingPatternParameter04.ts`)[https://github.com/Microsoft/TypeScript/blob/8952ed66b2e3ba78d64e8399384dd55ecfccabd8/tests/cases/conformance/es6/destructuring/emptyObjectBindingPatternParameter01.ts] whose baselines [seem fine](https://github.com/Microsoft/TypeScript/blob/8952ed66b2e3ba78d64e8399384dd55ecfccabd8/tests/baselines/reference/emptyObjectBindingPatternParameter01.js).

**Comment by DanielRosenwasser** [comment]
Actually @JsonFreeman, since the issue was when an identifier appeared in the initializer, *that's* what we should be testing. I'll amend my tests.

**Comment by DanielRosenwasser** [comment]
Done.

**Pull request merged**